src/code/fv-components/gem-set.js

```diff
@@ -0,0 +1,63 @@
+import React, {PropTypes} from 'react';
+import { getChallengeScores } from '../reducers/helpers/challenge-progress';
+
+export const GemView = ({challengeNum, score, highlight}) => {
+  let gem = score != null 
+              ? highlight
+                ? <div className="gem-outline-highlight">
+                    <div className={"gem-fill gem-fill-" + score}></div>
+                  </div>
+                : <div className={"gem-fill gem-fill-" + score}></div>
+              : <div className="gem-outline">
+                  <div className="gem-number-text">
+                    {challengeNum}
+                  </div>
+                </div>;
+  return (
+    <div className="gem-container">
+      {gem}
+    </div>
+  );
+};
+
+GemView.propTypes = {
+  challengeNum: PropTypes.number,
+  score: PropTypes.number,
+  highlight: PropTypes.bool
+};
+
+const GemSetView = ({level, mission, challenge, challengeCount, progress}) => {
+
+  let addAwardImage = (progressImages, pieces, gemNumber, score, currChallenge) => {
+    if (score > -1){
+      progressImages.push(<GemView score={score} highlight={currChallenge + 1 === gemNumber}/>);
+    } else {
+      progressImages.push(<GemView challengeNum={gemNumber}/>);
+    }
+    return progressImages;
+  };
+
+  let challengeScores = getChallengeScores(level, mission, challengeCount, progress),
+      progressImages = [];
+
+  for (let challengeNum = 0; challengeNum < challengeCount; challengeNum++) {
+    let gemNumber = parseInt(challengeNum) + 1;
+    progressImages = addAwardImage(progressImages, challengeCount, gemNumber, challengeScores[challengeNum], challenge);
```

Comment:
This is minor, but it feels odd to keep setting `progressImages` to a new array within a loop. More idiomatic would be something like
```
for (let challengeNum = 0; challengeNum < challengeCount; challengeNum++) {
  let highlight = challengeNum === currChallenge;
  progressImages.push(getAwardImage(challengeScores[challengeNum], highlight));
}
```
(Other minor simplifications: `parseInt` seems unnecessary; calculating the `highlight` to make `getAwardImage` more semantic; and `pieces` doesn't seem to be used in `addAwardImage`).